ui/components/app/snaps/snap-settings-page/snap-settings-renderer.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { FunctionComponent, useEffect, useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useLocation } from 'react-router-dom';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+
+import { deleteInterface } from '../../../../store/actions';
+import { Box, Text } from '../../../component-library';
+import {
+  BackgroundColor,
+  BlockSize,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import { SnapDelineator } from '../snap-delineator';
+import { getSnapMetadata } from '../../../../selectors';
+import { DelineatorType } from '../../../../helpers/constants/snaps';
+import { Copyable } from '../copyable';
+import { SnapUIRenderer } from '../snap-ui-renderer';
+import { useSnapSettings } from '../../../../hooks/snaps/useSnapSettings';
+import { decodeSnapIdFromPathname } from '../../../../helpers/utils/snaps';
+
+type SnapSettingsRendererProps = {
+  snapId: string;
+};
+
+export const SnapSettingsRenderer: FunctionComponent<
+  SnapSettingsRendererProps
+> = () => {
+  const { pathname } = useLocation();
+  const dispatch = useDispatch();
+  const t = useI18nContext();
+
+  const snapId = useMemo(() => decodeSnapIdFromPathname(pathname), [pathname]);
+
+  const { name: snapName } = useSelector((state) =>
+    getSnapMetadata(state, snapId),
+  );
+
+  if (!snapId) {
```

Comment:
This breaks the rules of React hooks since we might have two renders with not the same amount of hooks called

Reply:
What's you recommendation if we don't have a snap ID ?

Reply:
We can probably just move the `return null` down and support an undefined `snapId` in the `useSnapSettings` hook. WDYT?